java/ql/src/semmle/code/java/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -913,38 +1001,45 @@ LocalCallContext getLocalCallContext(CallContext ctx, DataFlowCallable callable)
   else result instanceof LocalCallContextAny
 }
 
+/**
+ * The value of a parameter at function entry, viewed as a node in a data
+ * flow graph.
+ */
+class ParameterNodeExt extends Node {
```

Comment:
I'm not particularly happy with the introduction of `ParameterNodeExt` and `ArgumentNodeExt` as cached aliases. I think we should rename these cached versions to `ParameterNode` and `ArgumentNode` and ensure that these cached versions get suitable public scope exposure to make them appear as `DataFlow::ParameterNode` and `DataFlow::ArgumentNode` instead of the current ones.
The language-specific definitions with those names then need to be dealt with in some way. Either hidden away in some module to ensure we don't get name clashes, or renamed, or replaced by predicates.

Reply:
The only problem with that is that languages may want to have more specific QL doc. Also, languages may want to add more predicates, for example in C# we have a `ParameterNode::getParameter()` predicate. Lastly, C# does not expose `ArgumentNode`.

Would it help if we renamed them to `ParameterNodeCached` and `ArgumentNodeCached`? (I know that the `Ext` naming is not good, but I wanted to keep it short.)

Reply:
That would also allow us to reuse the names for all the predicates that currently use the `Cached` postfix.